### DISCUSSION TITLE
HTTP AUTH endpoint to get all ACL rules of a user

### DISCUSSION THREAD

**Discussion by alexanderko** [nanomq/nanomq#1831] [Ideas]
Hello NanoMQ team, 

We have an application with lots of users and publish/subscribe permissions depending on their roles and OU they are added to. We are using EMQX on the Сloud and now considering deploying it On-Premise Windows 2016 servers using NanoMQ broker instead. 

Cloud version uses JWT auth with acl claim. For On-Premise HTTP AUTH module looks very promising and what we are needing. 

Our MQTT clients will subscribe to dozens of various topics on first connection and publish messages frequently. With this usage in mind it might be efficient to have one more HTTP AUTH endpoint in NanoMQ that will allow the HTTP AUTH Server to return all rules in one response in format similar to what EMQX have in [JWT acl claim](https://docs.emqx.com/en/emqx/latest/access-control/authn/jwt.html#new-format). The most effective it would be with having HTTP API allowing to update ACL for a specific user using the same payload structure.

Do you think such new features will be possible to see in NanoMQ in the near future?

**Comment by JaylinYu** [comment]
Yes, definitely could. But I cannot promise you the schedule, only if there is a paid contract of it.

**Reply by andriisly** [reply]
Hello @JaylinYu,

We would like to clarify whether the required HTTP AUTH endpoint was implemented in the recent updates? Or perhaps you know the approximate dates for implementation?

**Reply by JaylinYu** [reply]
Auth & ACL has already supported HTTP end point, but JWT is only for HTTP Server though. Not sure which one you mean.
 We don't have plan to add JWT token to HTTP ACL in opensource.

**Reply by JaylinYu** [reply]
I guess you want a new HTTP API for updating and reviewing the ACL rules. Apologize that, Currently we don't have enough human resources to add such an administrative feature to opensource ver
